uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/contract-entitlements.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-env amd */
+
+define([
```

Comment:
at the moment I think most of this file needs to change and can be improved, here is what I suggest
```javascript
function controller ($log, $q, HR_settings, AbsenceType, Contract) {
  $log.debug('Component: contract-entitlements');

  var vm = {};

  vm.absenceTypes = [];
  vm.contracts = [];

  (function init () {
    loadAbsenceTypes();
    loadContracts();
  })();

  function formatDate(date) {
    var dateFormat = HR_settings.DATE_FORMAT.toUpperCase(); //HR settings.DATE_FORMAT is the date format set by user
    
    return date ? moment(date).format(dateFormat) : null;
  }
  /**
   * Loads contracts
   *
   * @return {Promise}
   */
  function loadContracts () {
    return Contract.all({ contact_id: vm.contactId })
      .then(function (contracts) {
        vm.contracts = _.sortBy(contracts, function (contract) {
          return moment();
        }).map(function (contract) {
          return {
            position: contract.details.position,
            period_start_date: formatDate(contract.details.period_start_date),
            period_end_date: formatDate(contract.details.period_end_date)
          }
        });
      });
  }

  /**
   * Loads absence types
   *
   * @return {Promise}
   */
  function loadAbsenceTypes () {
    return AbsenceType.all()
      .then(function (absenceTypes) {
        vm.absenceTypes = absenceTypes;
      });
  }

  return vm;
}
```